File: 백준/Silver/10828. 스택/스택.py
import sys
input=sys.stdin.readline
stack=[]
N=int(input().rstrip())
for _ in range(N):
  M=input().rstrip()
  if("push" in M):
    stack.append(int(M[5:]))
  elif("pop" in M):
    if(len(stack)!=0):
      print(stack.pop())
    else:
      print(-1)
  elif(M=="size"):
    print(len(stack))
  elif(M=="empty"):
    if(len(stack)!=0):
      print(0)
    else:
      print(1)
  elif("top" in M):
    if(len(stack)!=0):
      print(stack[-1])
    else:
      print(-1)

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;

public class Main {
    public static void main(String[]args){
        Stack<Integer> stack= new Stack<>();
        Scanner scan= new Scanner(System.in);
        int N= scan.nextInt();
        for(int i=0;i<N; i++){
            String M= scan.next();
            if(M.equals("push")){
                int O= scan.nextInt();
                stack.push(O);
            }
            else if(M.equals("pop")){
                if (stack.isEmpty()) {
                    System.out.println(-1);
                } else
                    System.out.println(stack.pop());
            }
            else if(M.equals("size")){
                System.out.println(stack.size());
            }
            else if (M.equals("empty")) {
                if (stack.isEmpty()) {
                    System.out.println(1);
                }
                else {
                    System.out.println(0);
                }
            }
            else if (M.equals("top")) {
                if (stack.isEmpty()) {
                    System.out.println("-1");
                } else
                    System.out.println(stack.peek());
            }
        }
    }
}